.github/workflows/ci.yml

```diff
@@ -60,6 +60,8 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
+        with:
+          fetch-depth: 2
```

Comment:
this is needed for codecov to be able to detect correct commit SHA

> ==> GitHub Actions detected.
> ->  Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0

Reply:
could you set it to `0`?

Reply:
we could but its better not to, fetching only 1 commit is huge boost to time of git fetch/clone

this property is used to tell git how many commit we want to fetch